File: 0001-Two-Sum/Solutions/cpp/Solution02.cpp
/*
* @Author: Tony Liu
* @Date:   2020-02-16 22:52:34
* @Modified by:   Tony Liu
* @Last Modified time: 2020-02-16 22:58:50
*/

#include<vector>
#include<unordered_map>

using namespace std;

class Solution{
public:
	vector<int> twoSum(vector<int>& nums, int target){
		unordered_map<int,int> record;
		for(int i=0; i<nums.size(); i++)
			record[nums[i]] = i;

		for(int i=0;i<nums.size();i++){
			unordered_map<int,int>::iterator iter = record.find(target - nums[i]);
			if(iter != record.end() && iter->second !=i)
				return {i, iter->second};
		}

		throw invalid_argument("No Solution for Two Sum !");
	}
}

File: 0001-Two-Sum/Solutions/java/Solution02.java
/*
* @Author: Tony Liu
* @Date:   2020-02-16 23:04:19
* @Modified by:   Tony Liu
* @Last Modified time: 2020-02-23 21:37:24
*/

import java.util.HashMap;


class Solution02{

	public int[] twoSum(int[] nums,int target){

		HashMap<Integer,Integer> map = new HashMap<Integer,Integer>();
		for(int i=0;i<nums.length;i++)
			map.put(nums[i],i);


		for(int i=0; i<nums.length;i++){
			Integer index = map.get(target - nums[i]);
			if(index != null && index != i){
				return new int[] {i,index};
			}
		}

		throw new IllegalArgumentException("No Solution for Two Sum !");
	}

}